crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -888,6 +889,73 @@ fn handle_leading_function_with_decorators_comment(comment: DecoratedComment) ->
     }
 }
 
+/// Handles comments between `**` and the variable name in dict unpacking
+/// It attaches these to the appropriate value node
+///
+/// ```python
+/// {
+///     **  # comment between `**` and the variable name
+///     value
+///     ...
+/// }
+/// ```
+fn handle_dict_unpacking_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    match comment.enclosing_node() {
+        // TODO: can maybe also add AnyNodeRef::Arguments here, but tricky to test due to
+        // https://github.com/astral-sh/ruff/issues/5176
+        AnyNodeRef::ExprDict(_) => {}
+        _ => {
+            return CommentPlacement::Default(comment);
+        }
+    };
+    if let Some(following) = comment.following_node() {
+        let preceding_end = match comment.preceding_node() {
+            Some(preceding) => preceding.end(),
+            None => comment.enclosing_node().start(),
+        };
+        let mut tokens_before = SimpleTokenizer::new(
```

Comment:
`first_non_trivia_token_rev` might be easier

Reply:
The forward version is prefered over `rev` (when possible) because rev always needs to find the start of the line to assert that the token isn't part of a comment.

```
# the slash is not a continuation token / 
# The following slash is a continuation token
a + /
```